test/index.ts

```diff
@@ -1,25 +1,59 @@
-import { fastify } from 'fastify';
-import { runServer } from '../src/server';
-import dotenv from 'dotenv';
 import axios from 'axios';
+import * as chai from 'chai';
+import prismaInstance from '../src/db';
+import { closeServer, runServer } from '../src/server';
+import { UserReq } from '../src/user/user.model';
 
-dotenv.config({ path: '../.test.env' });
-const app = fastify();
-const port: number = process.env.PORT || 8081;
-const server = axios.create({
+const port: number = process.env.PORT ? parseInt(process.env.PORT, 10) : 8081;
+const endpoint = axios.create({
   baseURL: `http://localhost:${port}`,
 });
+const input: UserReq = {
+  name: 'Gabrielle',
+  email: 'gabi@gmail.com',
+  password: 'coxinha123',
+  birthDate: '02-12-2000',
+};
 
-before(() => {
-  app.get('/hello', () => {
-    return 'hello, world!';
-  });
-  runServer(app, port);
+before(async () => {
+  try {
+    await runServer(port);
+  } catch (error) {
+    console.error(error);
+    process.exit(1);
+  }
 });
 
 describe('Hello, world endpoint', () => {
-  it('return hello, world', async () => {
-    const response = await server.get('/hello');
-    console.log(response.data);
+  it(`return "hello, world!"`, async () => {
+    const response = await endpoint.get('/hello');
+    chai.expect(response.data).to.be.eq('hello, world!');
+  });
+});
+
+describe('Create user endpoint', () => {
+  it('Response and create sucess', async () => {
+    const response = await endpoint.post('/users', input);
+    chai.expect(response.status).to.be.deep.eq(201);
+    const userCreated = await prismaInstance.user.findUnique({
+      where: {
+        email: input.email,
+      },
+    });
+    //Comparação de usuário criado no banco com dados do input e response
+    chai.expect(userCreated?.id).to.be.deep.eq(response.data.id);
+    chai.expect(userCreated?.name).to.be.deep.eq(input.name);
+    chai.expect(userCreated?.email).to.be.deep.eq(input.email);
+    chai.expect(userCreated?.birthDate).to.be.deep.eq(input.birthDate);
+    chai.expect(userCreated?.password).to.not.be.deep.eq(input.password);
+    chai.expect(response.data.name).to.be.deep.eq(input.name);
+    chai.expect(response.data.email).to.be.deep.eq(input.email);
+    chai.expect(response.data.birthDate).to.be.deep.eq(input.birthDate);
+    chai.expect(response.data).to.not.be.deep.eq(input.password);
```

Comment:
O `to.be.deep.eq` para strings é a mesma coisa que o `to.be.eq`, porque string é um valor primitivo. O `to.be.deep.eq` serve para objetos:

```ts
expect(resposne.data).to.be.deep.eq({
  id: userCreated!.id,
  name: input.name,
  email: input.email,
  birthDate: input.birthDate,
});
```

Assim vc testa todo o objeto retornado no body, e inclusive o teste falha se tiver campos a mais ou a menos do que deveria, como deve ser.

Vale para todos os bodies de resposta que queremos testar.